qt/python/mantidqtinterfaces/mantidqtinterfaces/dns_powder_tof/data_structures/dns_file.py

```diff
@@ -0,0 +1,215 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2021 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+
+"""
+Class which loads and stores a single DNS datafile in a dictionary.
+"""
+
+import numpy as np
+from mantidqtinterfaces.dns_powder_tof.data_structures.object_dict import ObjectDict
+from mantidqtinterfaces.dns_powder_tof.helpers.file_processing import (load_txt, save_txt)
+
+
+class DNSFile(ObjectDict):
+    """
+    Class for reading, writing and storing data of a single dns datafile.
+    This is a dictionary, but can also be accessed like attributes.
+    """
+    def __init__(self, datapath, filename):
+        super().__init__()
+        self.new_format = self.read(datapath, filename)
+
+    def write(self, datapath, filename):
+        # mostly stolen form nicos
+        txt = ''
+        separator = "#" + "-" * 74 + "\n"
```

Comment:
Is the DNS file formate something we have produced ourselves? much of this looks like it would be better handled as a JSON file instead of a long txt file. and would be less vulnurable to user error.

Reply:
That's a good point - thanks! Briefly speaking, we don't produce the DNS file format. This format is generated by an "external" Nicos software and it has been used for many years at DNS as a format to store raw experimental data. The GUI takes (loads) ".d_dat" files as an input and then a user chooses the options for data reduction and then the corresponding algorithms are executed.   

To answer your question, the write() method here was included to be used for another (Powder Elastic) operational mode of the GUI that has not been submitted yet. In that mode, we had an option to perform interpolation of standard files for the case when standard and sample data are measured at different angles. And the interpolation algorithm would use the write() method to create "interpolated standard datafiles". However, recently we decided that generating "interpolated datafiles" is a dangerous approach and we won't use it in the Powder Elastic mode. So I will remove this method upon revision.